File: tritium-remote/examples/generic_query.rs
use simple_logger::SimpleLogger;
use std::env;

#[tokio::main]
async fn main() {
    // output *all* log messages, including from underlying transport
    SimpleLogger::new().env().init().unwrap();

    let auth_token =
        env::var("TRITIUM_AUTH_TOKEN").expect("TRITIUM_AUTH_TOKEN environment variable not set");

    let mut tritium = tritium_remote::connect(
        "ws://localhost:1234",
        &auth_token,
        Some("tritium-remote example - generic query".to_string()),
    )
    .await
    .expect("failed to connect");

    let document: &str = "
        query { 
            system { 
                serial
                hosts {
                    name
                    cpuCount
                } 
            }
        }
    ";

    log::info!("Document: {document}");

    let result = tritium.query(&document, None).await.expect("query failed");

    log::info!("Result: {result:?}");
}

File: py-tritium-remote/python/generic_query.py
#!/usr/bin/env python3

import os
import asyncio
import tritium_remote


async def main():
    auth_token = os.environ["TRITIUM_AUTH_TOKEN"]
    host = os.environ.get("TRITIUM_HOST", "localhost")

    print("connecting...")
    tritium = await tritium_remote.connect(
        f"ws://{host}:1234", auth_token, "Python tritium-remote example - list hosts"
    )

    document = """
        query { 
            system { 
                serial
                hosts {
                    name
                    cpuCount
                } 
            }
        }
    """

    response = await tritium.query(document, None)
    print("Response: ", response)


asyncio.run(main())
